xbmc/video/VideoDimensions.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "VideoDimensions.h"
+#include <algorithm>
+#include <limits>
+
+std::vector<SResolution> CVideoDimensions::resolutions =
+{
+  { "480", "SD", { { 0, 720 }, { 0, 480 } } },
+  // With invalid settings it's possible to end up with e.g. a 720x592 DVD rip
+  { "576", "SD", { { 0, 768 }, { 480, 600 } } },
+  // 720p can be anything from 960x720 (1.33:1) to 1280x464 (2.70:1, e.g. Ben Hur)
+  { "720", "HD", { { 768, 1280 }, { 460, 720 } } },
+  // Full-SBS 3D can be 3840x1080, some weird shit can be 1080x1040
```

Comment:
don't forget about Full-TAB which would be 1920x2160 (also in your test cases). Also there might be SBS or OU 4:3 SD material, like it can be found here: http://www.3dtv.at/Movies/Index_en.aspx (the skydive f.e., which currently has scaling issues IIRC, but that's a different story)